community/lucene-index/src/main/java/org/neo4j/kernel/api/impl/labelscan/reader/LuceneAllEntriesLabelScanReader.java

```diff
@@ -48,28 +49,23 @@ public LuceneAllEntriesLabelScanReader( BoundedIterable<Document> documents, Bit
     public Iterator<NodeLabelRange> iterator()
     {
         final Iterator<Document> iterator = documents.iterator();
-        return new Iterator<NodeLabelRange>()
+        return new IteratorWrapper<NodeLabelRange,Document>( iterator )
```

Comment:
We could, but I favor `IteratorWrapper` since it's only one object and generally simpler to create.